src/domain/safe/entities/schemas/transfer.schema.ts

```diff
@@ -0,0 +1,58 @@
+import { JSONSchemaType } from 'ajv';
+import { NativeTokenTransfer, TokenTransfer } from '../transfer.entity';
+
+export const transferSchema: JSONSchemaType<
+  TokenTransfer | NativeTokenTransfer
+> = {
+  type: 'object',
+  required: [],
+  oneOf: [
+    {
+      type: 'object',
+      properties: {
+        type: {
+          type: 'string',
+          enum: ['ERC721_TRANSFER', 'ERC20_TRANSFER'],
+        },
+        executionDate: { type: 'string' },
+        blockNumber: { type: 'number' },
+        transactionHash: { type: 'string' },
+        to: { type: 'string' },
+        from: { type: 'string' },
+        tokenId: { type: 'string' },
+        tokenAddress: { type: 'string' },
+      },
+      required: [
+        'type',
+        'executionDate',
+        'blockNumber',
+        'transactionHash',
+        'to',
+        'from',
+        'tokenId',
+        'tokenAddress',
```

Comment:
TokenAddress seems to be optional following the openapi documentation.
https://safe-transaction.mainnet.gnosis.io/
https://github.com/safe-global/safe-transaction-service/blob/master/safe_transaction_service/history/serializers.py#L720